draft-ietf-tls-ticketrequests.md

```diff
@@ -124,32 +146,71 @@ Clients MAY send this extension in ClientHello. It contains the following struct
 
 ~~~
 struct {
-    uint8 count;
-} TicketRequestContents;
-~~~
+    uint8 new_session_count;
+    uint8 resumption_count;
+} ClientTicketRequest;
 
-count
-: The number of tickets desired by the client.
+struct {
+    uint8 expected_count;
+} ServerTicketRequest;
 
-Clients can use TicketRequestContents.count to indicate the number of tickets
-they would prefer to receive. Servers SHOULD NOT send more tickets than
-TicketRequestContents.count, as clients will most likely discard any additional
-tickets. Servers SHOULD additionally place a limit on the number of tickets
-they are willing to send, to save resources. Therefore, the number of
-NewSessionTicket messages sent will typically be the minimum of the server's self-imposed
-limit and TicketRequestContents.count.
+struct {
+    select (Handshake.msg_type) {
+        case client_hello: ClientTicketRequest;
+        case encrypted_extensions: ServerTicketRequest;
+    }
+} TicketRequestContents;
+~~~
 
-A server that supports ticket requests MAY echo the "ticket_request" extension in the
-EncryptedExtensions message. If present, it contains a TicketRequestContents structure, where
-TicketRequestContents.count indicates the number of tickets the server expects to send to the client.
+new_session_count
+: The number of tickets desired by the client when the server chooses to
+negotiate a fresh session (full handshake).
+
+resumption_count
+: The number of tickets desired by the client when the server is willing to
+resume using the presented ticket.
+
+expected_count
+: The number of tickets the server expects to send in response to a client
+ticket_request extension.
+
+A client starting a fresh connection SHOULD set new_session_count to the desired
+number of session tickets and resumption_count to 0.
+Once a client's ticket cache is primed, a resumption_count of 1 is a
+good choice that allows the server to replace each ticket with a fresh ticket,
+without over-provisioning the client with excess tickets. However, clients
+which race multiple connections and place a separate ticket in each will
+ultimately end up with just the tickets from a single resumed session.
+In that case, clients SHOULD send a resumption_count equal to the number
+of sessions they are attempting in parallel.
```

Comment:
This is advisory text, so I would prefer that you avoid SHOULD.  Particularly for the resumption_count, which could equally be N (as it cannot apply in the absence of resumption, the value doesn't matter).

Reply:
It was originally: 

```
In that case, a resumption_count commensurate with the number of parallel sessions would be used.
```

@DavidSchinazi, do you care strongly if we revert this?

Reply:
In that case I'd prefer something like `In that case, clients can send a resumption_count equal to the number of sessions they are attempting in parallel.` - I prefer active to passive, also I don't like the word commensurate :)

Reply:
That works for me!